igcollect/linux_disklatency.py

```diff
@@ -107,5 +77,46 @@ def main():
             print(template.format(mode, k, int(v)))
 
 
+def get_path(args) -> str:
+    if args.path is not None:
+        return args.path
+
+    # no --path is specified -> use libvirt to get the storage volume
+    from libvirt import (
+        open as libvirt_open,
+        libvirtError,
+    )
+
+    if args.storagepool is None or args.storagevol is None:
+        raise ArgumentError('if no --path is set you have to set --storagepool AND --storagevol')
+
+    try:
+        conn = libvirt_open('qemu:///system')
+    except libvirtError as e:
+        raise Exception(
```

Comment:
You can chain exceptions with `raise Exception(...) from e`. With that you still have the original exception in there.